_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,236 @@
+
+## Expectation
+
+Each target element is either the [row header](https://www.w3.org/TR/html-aria/#index-aria-rowheader) or the [column header](https://www.w3.org/TR/html-aria/#index-aria-columnheader) for data [cells](https://www.w3.org/TR/html50/tabular-data.html#concept-cell) within the same [table](https://www.w3.org/TR/html50/tabular-data.html#concept-table).
+
+## Assumptions
+
+_There are currently no assumptions._
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 1.3.1: Information and relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
+- [H43: Using id and headers attributes to associate data cells with header cells in data tables](https://www.w3.org/WAI/WCAG21/Techniques/html/H43)
+- [Forming relationships between data cells and header cells](https://html.spec.whatwg.org/multipage/tables.html#header-and-data-cell-semantics)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+Each `th` element has corresponding cells within the same `table`.
+
+```html
+<table>
+	<tr>
+		<th>Time</th>
+	</tr>
+	<tr>
+		<td>05:41</td>
+	</tr>
+</table>
+```
+
+#### Passed Example 2
+
+Each `th` element has corresponding cells within the same `table`. Here the cells span multiple columns.
+
+```html
+<table>
+  <thead>	
+    <tr>
+      <th>Projects</th>
+      <th>Exams</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td colspan="2">15%</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+#### Passed Example 3
+
+The `th` element has corresponding cells within the same `table`, by usage of `headers` attribute.
+
+```html
+<table>
+  <tr> 
+    <td headers="header1">
+      Projects
+    </td>
+    <th id="header1">
+      Projects
+    </th> 
+  </tr>
+</table>
+```
+
+#### Passed Example 4
+
+The `th` element has corresponding cells within the same `table`, by usage of `aria-labelledby` attribute.
+
+```html
+<table>
+	<tr>
+    <th id="hd2">Time</th>
+    <td aria-labelledby="hd2">05:44</td>
+	</tr>
+</table>
+```
+
+#### Passed example 5
+
+Each `th` element has corresponding cells within the same `table`, by usage of `scope` attribute.
+
+```html
+<table>
+  <caption>Hogwarts Houses stars</caption>
+  <tr>
+      <th scope="col">Name</th>
+      <th scope="col">House</th>
+  </tr>
+  <tr>
+      <th scope="row">Harry Potter</th>
+      <td>Gryffindor</td>
+  </tr>
+  <tr>
+      <th scope="row">Cedric Diggory</th>
+      <td>Hufflepuff</td>
+  </tr>
+</table>
+```
+
+#### Passed example 6
+
+Each `th` element has corresponding cells within the same `table`.
+
+```html
+<table>
+	<caption>Hogwarts Houses stars</caption>
+	<tr>
+			<th>Name</th>
+			<th>House</th>
+	</tr>
+	<tr>
+			<th>Cho Chang</th>
+			<td>Ravenclaw</td>
+	</tr>
+	<tr>
+			<th>Severus Snape</th>
+			<td>Slytherin</td>
+	</tr>
+</table>
+```
+
+### Failed
```

Comment:
I think we could use another failed example, with a bit more "complexity", i.e., with `td` cells also. Something like:

```html
<table>
    <tr>
        <th>Header</th>
        <td>Data 1</td>
    </tr>
    <tr>
        <td>Data 2</td>
        <td>Data 3</td>
    </tr>
</table>
```

Reply:
The rule only looks at `th` elements, and for the above example it satisfies the expectation, so this would not be a fail.

Reply:
Yes, you're completely right!